target/linux/layerscape/image/armv8_64b.mk

```diff
@@ -86,6 +86,12 @@ define Device/ls1012afrwy
     ls-append-dtb $$(DEVICE_DTS) | pad-to 16M | \
     append-kernel | pad-to $(LS_SD_ROOTFSPART_OFFSET)M | \
     append-rootfs | check-size $(LS_SD_IMAGE_SIZE)
+  IMAGE/sysupgrade.bin := \
+    ls-clean | \
+    ls-append-sdhead $(1) | pad-to 16M | \
+    append-kernel | pad-to $(LS_SD_ROOTFSPART_OFFSET)M | \
+    append-rootfs | check-size $(LS_SD_IMAGE_SIZE) | append-metadata
+  SUPPORTED_DEVICES := fsl,ls1012a-frwy-sdboot
```

Comment:
I don't find these -sdboot suffixed values anywhere else, particularly not in any compatible as you partially reuse the DTS files anywhere. Is this actually working for sysupgrade then?

Reply:
In OpenWrt, the board name info is created in by /lib/preinit/02_sysinfo.
You can find below patch of this pull request to fix up the board name for SD card boot.
layerscape: fix board name for SD card boot

Reply:
Okay, this was the missing link I was looking. While I don't really like that approach conceptually, it appears to do the job.

Can you explain to me how the filtering works there? How do you make sure -sdboot is only added for sdboot variants, and not for all devices.

(Despite, I'd have used a higher number for that script, as sorting between 02_sysinfo and 02_sysinfo_fixup is not really obvious).